knotx-adapter/knotx-adapter-common/src/main/java/io/knotx/adapter/common/http/HttpClientFacade.java

```diff
@@ -18,51 +18,46 @@
 import io.knotx.adapter.common.exception.AdapterServiceContractException;
 import io.knotx.adapter.common.exception.UnsupportedServiceException;
 import io.knotx.adapter.common.placeholders.UriTransformer;
-import io.knotx.adapter.common.post.UrlEncodedBodyBuilder;
 import io.knotx.dataobjects.AdapterRequest;
 import io.knotx.dataobjects.ClientRequest;
 import io.knotx.dataobjects.ClientResponse;
 import io.knotx.http.AllowedHeadersFilter;
 import io.knotx.http.MultiMapCollector;
-import io.netty.handler.codec.http.HttpResponseStatus;
 import io.vertx.core.http.HttpMethod;
 import io.vertx.core.json.JsonObject;
 import io.vertx.core.logging.Logger;
 import io.vertx.core.logging.LoggerFactory;
 import io.vertx.rxjava.core.MultiMap;
 import io.vertx.rxjava.core.buffer.Buffer;
-import io.vertx.rxjava.core.http.HttpClient;
-import io.vertx.rxjava.core.http.HttpClientRequest;
-import io.vertx.rxjava.core.http.HttpClientResponse;
+import io.vertx.rxjava.ext.web.client.HttpRequest;
+import io.vertx.rxjava.ext.web.client.HttpResponse;
+import io.vertx.rxjava.ext.web.client.WebClient;
 import java.util.List;
 import java.util.Optional;
 import java.util.regex.Pattern;
 import org.apache.commons.lang3.tuple.Pair;
-import rx.Observable;
+import rx.Single;
 
 public class HttpClientFacade {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientFacade.class);
   private static final String PATH_PROPERTY_KEY = "path";
-  private static final ClientResponse INTERNAL_SERVER_ERROR_RESPONSE = new ClientResponse()
-      .setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
 
   private final List<ServiceMetadata> services;
 
-  private final HttpClient httpClient;
+  private final WebClient webClient;
 
-  public HttpClientFacade(HttpClient httpClient, List<ServiceMetadata> services) {
-    this.httpClient = httpClient;
+  public HttpClientFacade(WebClient webClient, List<ServiceMetadata> services) {
+    this.webClient = webClient;
     this.services = services;
   }
 
-  public Observable<ClientResponse> process(AdapterRequest message, HttpMethod method) {
-    return Observable.just(message)
-        .doOnNext(this::validateContract)
+  public Single<ClientResponse> process(AdapterRequest message, HttpMethod method) {
+    return Single.just(message)
+        .doOnSuccess(this::validateContract)
```

Comment:
If I understand correctly `defaultIfEmpty` was triggered only when `flatMap` in line `64` returned no events.
`validateContract` throws exception, sot whole stream emmits `onError` - this flow is still true with `Single`.
I can't see other way than `onSuccess` when we create a `Single` with `just` :)

According to emitting `defaultIfEmpty` after `wrapResponse`:
I'm wondering when it may happen that nothing is emitted on this level... Is that possible with some timeout/network error? Such events will rather emit `error` than nothing.
Do you see any case when `defaultIfEmpty` could be triggered?